src/nested_ragged_tensors/ragged_numpy.py

```
@@ -989,124 +1217,172 @@
+                curr_st = S.start
+                curr_end = S.stop
+
+                if curr_st is not None:
+                    st_i += curr_st
+                    if end_i is not None:
+                        end_i += curr_st
+                if curr_end is not None:
+                    if end_i is None:
+                        end_i = curr_end
+                    else:
+                        end_i = min(end_i, curr_end)
```

Ruff

Use ternary operator `end_i = curr_end if end_i is None else min(end_i, curr_end)` instead of `if`-`else`-block

Replace `if`-`else`-block with `end_i = curr_end if end_i is None else min(end_i, curr_end)`

(SIM108)